modules/metastore/src/Entity/ResourceMapping.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\metastore\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\metastore\ResourceMappingInterface;
+
+/**
+ * Defines the resource mapping entity class.
+ *
+ * Used as storage by \Drupal\metastore\ResourceMapper.
```

Comment:
Suggestion... 
ResourceMappings store particular "perspectives" on a data file resource. The "source" perspective will be a URL to the original location of the resource. Local and LocalUrl perspectives record local copies of remote files for use by the DKAN datastore. A ResourceMapping entity has three levels of definition - the identifier, which is hash of the source URL; the version, which is a timestamp of when the source URL was accessed/downloaded; and the perspective itself, which will be one of those three values. These mappings allow DKAN to work with local copies of remote files, particularly for import into the datastore, and keep track of which remote files they represent.